test/code/providers/runas_provider/runasprovider_test.cpp

```diff
@@ -86,6 +91,17 @@ class SCXRunAsProviderTest : public CPPUNIT_NS::TestFixture
     CPPUNIT_TEST_SUITE_END();
 
 public:
+    
+    int RunCommand(std::wstring command)
+    {
+    	std::istringstream input;
+    	std::ostringstream output, error;
+    	int code = SCXCoreLib::SCXProcess::Run(command, input, output, error);
+    	if(error.str().length() > 0)
+    		std::cout<<"SXProcess::Run error: "<<error.str()<<std::endl;
+    	return code;
+    }
```

Comment:
That's a great idea to have a RunCommand to isolate this. A few suggestions:

1. In addition to checking the length of stderr (which you did, you should also check the return code (i.e. if (code != 0 || error.str().length() > 0) ...

2. You could do a CPPUNIT_ASSERT in here. If you do that, that will cause the caller to abort as well, and then you could make this function void and get rid of testing on return. A failure would simply cause a test failure. That would eliminate a number of changes here.

3. A disadvantage to nested ASSERTs is that it's hard to tell who called. So what I've done in the past is take a test name as parameter two, and then use an assertion with a message to show who the caller was. You could do something like:

CPPUNIT_ASSERT_EQUAL_MESSAGE(L"Unpexpected stdout error '<stdout_text>' running command <cmd>, called from <test>, error: 'errorText', string(""), error.str());
CPPUNIT_ASSERT_EQUAL_MESSAGE(L"Failure running command <cmd>, called from <test>", 0, code);

You'll obviously need to do something to build the string (perhaps use a stringstream, or use the StrUtil functions to build it. Easier to use a stringstream for more complex output, like above.

I like checking the stderr output first, as this is often more descriptive than a non-zero return code.

Again, doing this allows you to make RunCommand() void, meaning you don't need to check further down the road.

Finally, I'm not 100% sure what happens if you ASSERT in tearDown(). Try it and, if it's really unpleasant, then you could only do the above CPPUNIT assertions if <test> != "tearDown").